scene-lib/src/annotations/io/IndexFileParser.java

```diff
@@ -560,19 +566,13 @@ private ScalarAFT parseScalarAFT() throws IOException, ParseException {
             if (abbreviate) {
               int i = name.lastIndexOf('.');
               if (i >= 0) {
-                String baseName = name.substring(i+1);
-                Set<String> set1 = scene.imports.get(name);
-                Set<String> set2 = scene.imports.get(baseName);
-                if (set1 == null) {
-                  set1 = new TreeSet<String>();
-                  scene.imports.put(name, set1);
-                }
-                if (set2 == null) {
-                  set2 = new TreeSet<String>();
-                  scene.imports.put(name, set2);
+                Set<String> importSet = scene.imports.get(annotationFullyQualifiedName);
+                if (importSet == null) {
+                  importSet = new TreeSet<String>();
+                  scene.imports.put(annotationFullyQualifiedName, importSet);
                 }
-                set1.add(name);
-                set2.add(name);
+                importSet.add(name);
```

Comment:
Hi Mernst, I think `importSet` is just a set manages imports for a specific annotation, and thus there is no need the checking `name` existence in an `importSet` before adding this `name` to it. If `name` already exists in `importSet`, calling `importSet.add(name)` actually will not change the set.

`scene.imports` is the map that manages the mapping from annotation name to its imports, and since we use fully-qualified annotation name as map key, so adding new mapping for a different annotation in this map will not overwrite other annotations' mapping.